examples/spark/lgbm-parallel.py

```diff
@@ -0,0 +1,245 @@
+            "roles": train_sdataset.roles,
+            "task_type": dataset.task_type,
+            "target": dataset.roles["target"]
+        }
+
+        with open(self.metadata_path, "wb") as f:
+            pickle.dump(metadata, f)
+
+        logger.info(f"Dataset {self.dataset_name} has been prepared.")
+
+    @property
+    def train_dataset(self) -> DataFrame:
+        return self.spark.read.parquet(self.train_path)
+
+    @property
+    def test_dataset(self) -> DataFrame:
+        return self.spark.read.parquet(self.test_path)
+
+    @property
+    def metadata(self) -> Dict[str, Any]:
+        with open(self.metadata_path, "rb") as f:
+            return pickle.load(f)
+
+    def train_model(self, fold: int) -> Tuple[int, float]:
+        logger.info(f"Starting to train the model for fold #{fold}")
+
+        train_df = self.train_dataset
+        test_df = self.test_dataset
+        md = self.metadata
+        task_type = md["task_type"]
+
+        train_df.sql_ctx.sparkSession.sparkContext.setLocalProperty("spark.scheduler.mode", "FAIR")
+
+        prediction_col = 'LightGBM_prediction_0'
+        if task_type in ["binary", "multiclass"]:
+            params["rawPredictionCol"] = 'raw_prediction'
+            params["probabilityCol"] = prediction_col
+            params["predictionCol"] = 'prediction'
+            params["isUnbalance"] = True
+        else:
+            params["predictionCol"] = prediction_col
+
+        if task_type == "reg":
+            params["objective"] = "regression"
+            params["metric"] = "mse"
+        elif task_type == "binary":
+            params["objective"] = "binary"
+            params["metric"] = "auc"
+        elif task_type == "multiclass":
+            params["objective"] = "multiclass"
+            params["metric"] = "multiclass"
+        else:
+            raise ValueError(f"Unsupported task type: {task_type}")
+
+        if task_type != "reg":
+            if "alpha" in params:
+                del params["alpha"]
+            if "lambdaL1" in params:
+                del params["lambdaL1"]
+            if "lambdaL2" in params:
+                del params["lambdaL2"]
+
+        assembler = VectorAssembler(
+            inputCols=list(md['roles'].keys()),
+            outputCol=f"LightGBM_vassembler_features",
+            handleInvalid="keep"
+        )
+
+        lgbm_booster = LightGBMRegressor if task_type == "reg" else LightGBMClassifier
+
+        lgbm = lgbm_booster(
+            **params,
+            featuresCol=assembler.getOutputCol(),
+            labelCol=md['target'],
+            validationIndicatorCol='is_val',
+            verbosity=1,
+            useSingleDatasetMode=True,
+            isProvideTrainingMetric=True,
+            chunkSize=4_000_000,
+            useBarrierExecutionMode=True,
+            numTasks=2,
+            numThreads=2
+        )
+
+        if task_type == "reg":
+            lgbm.setAlpha(0.5).setLambdaL1(0.0).setLambdaL2(0.0)
```

Comment:
why constants? and not parameters?

Reply:
This part directly follows LightAutoML library (which is a desired behavior).
As I understand, these parameters are identified as good in global sense and thus not a subject for change.
Won't fix.